modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java

```diff
@@ -2623,7 +2623,7 @@ protected void updateModelForComposedSchema(CodegenModel m, Schema schema, Map<S
 
         // TODO revise the logic below to set discriminator, xml attributes
         if (supportsInheritance || supportsMixins) {
-            m.allVars = new ArrayList<>();
+            m.allVars = m.allVars == null ? new ArrayList<>() : m.allVars;
```

Comment:
can you please explain this change?

looks like this change impacts the spring codegen as well.

Reply:
> can you please explain this change?

It was obvious mistake to assume that allVars is null and need to assigned to ArrayList. Well, it was null in most cases, but in some rare ones it's already populated with important data and shall not be cleared by re-instantiation.

> looks like this change impacts the spring codegen as well.

yes, it may. But I've checked, didn't look like a problem year ago.